qa/integration-tests/src/test/java/io/zeebe/broker/it/clustering/RestoreTest.java

```diff
@@ -124,19 +126,70 @@ public void shouldKeepPositionsConsistent() {
     }
   }
 
+  // Regression test added for https://github.com/zeebe-io/zeebe/issues/6438
+  @Test
+  public void shouldBecomeLeaderAfterRestoreFromSnapshot() {
+    // given
+    clusteringRule.stopBrokerAndAwaitNewLeader(2);
+    writeManyEventsUntilAtomixLogIsCompactable();
+    clusteringRule.getClock().addTime(SNAPSHOT_PERIOD);
+    clusteringRule.waitForSnapshotAtBroker(getLeader());
+
+    // when
+
+    // Bring the stopped broker back - So that it has receives the snapshot from the leader
+    clusteringRule.restartBroker(2);
+    clusteringRule.waitForSnapshotAtBroker(clusteringRule.getBroker(2));
+
+    // After stopping 0, writing more events, stopping 1, restarting 0 only 2 can become leader.
```

Comment:
I like that this comment tries to explain why, but I think we can make it clearer.

```suggestion
    // writing more events after stopping 0 ensures that 2 will become leader since once we stop 1, 2 will have the longest log between it and 0
```